internal/resource/config/serversettings/logsettings/server_settings_log_settings_data_source.go

```diff
@@ -0,0 +1,136 @@
+package serversettingslogsettings
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/datasource/common/id"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &serverSettingsLogSettingsDataSource{}
+	_ datasource.DataSourceWithConfigure = &serverSettingsLogSettingsDataSource{}
+)
+
+// ServerSettingsLogSettingsDataSource is a helper function to simplify the provider implementation.
+func NewServerSettingsLogSettingsDataSource() datasource.DataSource {
+	return &serverSettingsLogSettingsDataSource{}
+}
+
+// serverSettingsLogSettingsDataSource is the datasource implementation.
+type serverSettingsLogSettingsDataSource struct {
+	providerConfig internaltypes.ProviderConfiguration
+	apiClient      *client.APIClient
+}
+
+type serverSettingsLogSettingsDataSourceModel struct {
+	Id            types.String `tfsdk:"id"`
+	LogCategories types.Set    `tfsdk:"log_categories"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *serverSettingsLogSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes the settings related to server logging.",
+		Attributes: map[string]schema.Attribute{
+			"log_categories": schema.SetNestedAttribute{
+				Description: "The log categories defined for the system and whether they are enabled. On a PUT request, if a category is not included in the list, it will be disabled.",
```

Comment:
You could remove the sentence about the PUT request, since it's not relevant to data sources. Same for the other descriptions that refer to PUTs.